### PR TITLE
Component tests & Fixes

### LINKED ISSUE #325: Investigate transaction size estimation
From clients we obtain a lot of errors when broadcasting such as `min relay fee not met, 650 < 653 (code 66)\`

﻿

### PR DESCRIPTION
**Pull request by altafan**
This adds component test for wallet and operator services:
* for the former, tests validate the init wallet flow for a new wallet from scratch, a restarted daemon and a restored wallet.
* for the latter, there's a test that validates the flow of creating and funding fee/market accounts. As soon as we support blinding via blinders, we should add also a test to validate the withdrawal from market.

BONUS:
* Fix error messages of init wallet when it's not yet synced/initialized.
* DbManager has been renamed DbManager and repos are now created and retrieved through it.
* WalletService.DepositMarket now returns a slice of AddressAndBlindingKey instead of only addresses. It's responsibility of the grpc interface to return only addresses.
* Add `BlinderManager` to be able to mock unblinding of outputs
* Add GetTransaction to explorer.Service
* Add methods to blind with blinding data in pkg/wallet. TODO: use them to let the daemon blind txs without unblinding inputs again (it already has them as unblinded unspent).
* Drop MinRelayFee and refactor tx size estimation.

This closes #325.

Please @tiero review this.

### CODE REVIEW

**Comment by altafan** [comment]
@sekulicd you may want to look at how `EstimateTxSize` has been changed.

File: pkg/wallet/estimation.go [hunk 1]
```diff
@@ -110,13 +113,21 @@ func calcTxBaseSize(
...
 
-func calcTxWitnessSize(inScriptTypes, outScriptTypes []int) int {
+func calcTxWitnessSize(
+	inScriptTypes, inAuxiliaryWitnessSize,
+	outScriptTypes []int,
+) int {
 	insSize := 0
+	auxCount := 0
 	for _, scriptType := range inScriptTypes {
-		if scriptType > P2MS {
```
Comment (tiero) [inline pkg/wallet/estimation.go]: is this breaking any of other project may using this package?

Comment (altafan) [reply pkg/wallet/estimation.go]: the estimation function wasn't exposed before this or #322, so other projects would not be affected by this. Anyway, I changed this specific check to increase the precision of estimation for txs with ins/outs of type p2sh(p2wsh) or p2wsh.

**Review by tiero — APPROVED** [review]

File: pkg/wallet/util.go [hunk 2]
```diff
@@ -161,9 +161,10 @@ func addInsAndOutsToPset(
 	return ptx.ToBase64()
 }
 
-func extractScriptTypesFromPset(ptx *pset.Pset) ([]int, []int, []int, []int) {
+func extractScriptTypesFromPset(ptx *pset.Pset) ([]int, []int, []int, []int, []int) {
```
Comment (sekulicd) [inline pkg/wallet/util.go:164]: should we move this to esitamtion.go ?

File: pkg/wallet/util.go [hunk 3]
```diff
@@ -213,13 +222,26 @@ func extractScriptTypesFromPset(ptx *pset.Pset) ([]int, []int, []int, []int) {
...
 			outScriptTypes[i] = P2WSH
 		}
 	}
 
-	return inScriptTypes, outScriptTypes, inAuxiliaryP2ShSize, outAuxiliaryP2ShSize
+	return inScriptTypes, inAuxiliaryRedeemScriptSize, inAuxiliaryWitnessSize,
+		outScriptTypes, outAuxiliaryRedeemScriptSize
+}
+
+func calcWitnessSizeFromRedeemScript(script []byte) int {
```
Comment (sekulicd) [inline pkg/wallet/util.go:237]: should we move this to esitamtion.go ?

**Review by sekulicd — COMMENTED** [review]
we should use camel case for constants also.

File: pkg/wallet/util.go (hunk 2 above)
Comment (altafan) [reply pkg/wallet/util.go:164]: I put these functions in util because they're used in transasction.go

**Pull request merged**